clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/SubmitTitusJob.java

```diff
@@ -98,7 +90,7 @@ public Result apply(@NotNull SubmitTitusJobCommand command, @NotNull Saga saga)
         retrySupport.retry(
             () -> {
               try {
-                return titusClient.submitJob(submitJobRequest);
+                return titusClient.submitJob(submitJobRequest.withJobName(nextServerGroupName[0]));
```

Comment:
This class is so upsetting, but there was already enough change in this PR. I want to come back and make this class not super gross.